pkg/api/defaults.go

```diff
@@ -202,6 +202,12 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 				}
 			}
 		}
+
+		if o.KubernetesConfig.NetworkPlugin == NetworkPluginAzure {
+			if o.KubernetesConfig.NetworkMode == "" {
+				o.KubernetesConfig.NetworkMode = NetworkModeTransparent
```

Comment:
sorry for a dumb question but what actually actions on network mode? I couldn't find it in my novice sluething. I just found this in cse.


if [[ "${NETWORK_POLICY}" == "calico" ]]; then
--
284 | sed -i 's#"mode":"bridge"#"mode":"transparent"#g' $CNI_CONFIG_DIR/10-azure.conflist

Reply:
https://github.com/Azure/aks-engine/blob/1dd327ad7697d4fa43d2bb86adf6679689ab7e84/parts/k8s/cloud-init/artifacts/cse_config.sh#L288

is it this one?

Reply:
@weng271190436 has it right

Basically, the Azure CNI release artifact comes with a default static config, and we simply extract it, then sed replace `"bridge"` with `"transparent"` if the api model says to do so.